### PR TITLE
Fix HFTrainer overloads

### PR DESCRIPTION
**Pull request by kylesayrs**
## Purpose ##
* Fix failing tests in `tests/llmcompressor/transformers/finetune/test_finetune_no_recipe_custom_dataset.py`

## Background ##
* The HFTrainer function contracts were changed in the most recent transformers release 4.46.0 ([offending commit](https://github.com/huggingface/transformers/commit/6ba31a8a94bf7cfeaf59ffc3bc9e0b0cd3e25788#diff-ed55888e6665791fe92cc8fc0c499da54f4ace6738551cd9a2591881cda076deR3603))

## Changes ##
* Change mixin overloads to use the same function contract as the original functions
* This change is backwards compatible with earlier versions of transformers since the new arguments are kwargs

## Testing ##
* `tests/llmcompressor/transformers/finetune/test_finetune_no_recipe_custom_dataset.py` now passes

### CODE REVIEW

File: src/llmcompressor/transformers/finetune/session_mixin.py [hunk 1]
```diff
@@ -286,7 +293,7 @@ def compute_loss(
 
         # TODO: do we need these model signature columns?
         inputs = {k: inputs[k] for k in inputs if k in self._signature_columns}
-        loss = super().compute_loss(model, inputs, return_outputs=return_outputs)
+        loss = super().compute_loss(model, inputs, return_outputs, num_items_in_batch)
```
Comment (mgoin) [inline src/llmcompressor/transformers/finetune/session_mixin.py:296]: It would be nice to keep named args here

**Review by mgoin — DISMISSED** [review]

File: src/llmcompressor/transformers/finetune/session_mixin.py (hunk 1 above)
Comment (kylesayrs) [reply src/llmcompressor/transformers/finetune/session_mixin.py:296]: I’d prefer to keep the call more concise in the case that the reader knows exactly what the function call is, unless there are any other good reasons

**Review by rahul-tuli — APPROVED** [review]

File: src/llmcompressor/transformers/finetune/session_mixin.py (hunk 1 above)
Comment (dsikka) [reply src/llmcompressor/transformers/finetune/session_mixin.py:296]: I think we should use arg names to help with readability and in the event that one of them is removed, the error will indicate that it is no longer relevant

**Review by dsikka — DISMISSED** [review]

**Review by rahul-tuli — DISMISSED** [review]

**Comment by rahul-tuli** [comment]
Should we add *args, **kwargs and pass them to the super functions, to protect against future changes like this?

File: src/llmcompressor/transformers/finetune/session_mixin.py [hunk 2]
```diff
@@ -265,12 +268,16 @@ def training_step(
         self._check_super_defined("training_step")
 
         callbacks.batch_start(batch_data=inputs)
-        model_outputs = super().training_step(model, inputs)
+        model_outputs = super().training_step(model, inputs, num_items_in_batch)
```
Comment (mgoin) [inline src/llmcompressor/transformers/finetune/session_mixin.py:271]: nit: arg names here too :)

**Review by mgoin — DISMISSED** [review]

**Comment by kylesayrs** [comment]
@rahul-tuli Since both functions require some information from the inputs, I think I'd prefer to expose all the arguments. There are tradeoffs either way w.r.t. erroring when arguments change

**Review by dsikka — APPROVED** [review]

**Pull request merged**